ci-operator/step-registry/cucushift/upgrade/tolatest/cucushift-upgrade-tolatest-commands.sh

```diff
@@ -0,0 +1,681 @@
+    local version state
+    version=$(oc get clusterversion/version -o jsonpath='{.status.history[0].version}')
+    state=$(oc get clusterversion/version -o jsonpath='{.status.history[0].state}')
+    if [[ ${version} == "${TARGET_VERSION}" && ${state} == "Completed" ]]; then
+        echo "History check PASSED, cluster is now upgraded to ${TARGET_VERSION}" && return 0
+    else
+        echo >&2 "History check FAILED, cluster upgrade to ${TARGET_VERSION} failed, current version is ${version}, exiting" && return 1
+    fi
+}
+
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Get the target upgrades release, by default, OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE is the target release
+# If it's serial upgrades then override-upgrade file will store the release and overrides OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE
+# upgrade-edge file expects a comma separated releases list like target_release1,target_release2,...
+export TARGET_RELEASES=("${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}")
+if [[ -f "${SHARED_DIR}/upgrade-edge" ]]; then
+    release_string="$(< "${SHARED_DIR}/upgrade-edge")"
+    # shellcheck disable=SC2207
+    TARGET_RELEASES=($(echo "$release_string" | tr ',' ' '))
+fi
+echo "Upgrade targets are ${TARGET_RELEASES[*]}"
+
+export OC="run_command_oc"
+
+# Target version oc will be extract in the /tmp/client directory, use it first
+mkdir -p /tmp/client
+export OC_DIR="/tmp/client"
+export PATH=${OC_DIR}:$PATH
+
+for target in "${TARGET_RELEASES[@]}"
+do
+    export TARGET="${target}"
+    # we need this DUMMY_TARGET_VERSION from ci config to download oc and do some pre-check
+    DUMMY_TARGET_VERSION="$(env "NO_PROXY=*" "no_proxy=*" oc adm release info "${TARGET}" --output=json | jq -r '.metadata.version')"
+    extract_oc
+
+    SOURCE_VERSION="$(oc get clusterversion --no-headers | awk '{print $2}')"
+    SOURCE_MINOR_VERSION="$(echo "${SOURCE_VERSION}" | cut -f2 -d.)"
+    export SOURCE_VERSION
+    export SOURCE_MINOR_VERSION
+    echo -e "Source release version is: ${SOURCE_VERSION}\n"
+
+    TARGET_MINOR_VERSION="$(echo "${DUMMY_TARGET_VERSION}" | cut -f2 -d.)"
+    export TARGET_MINOR_VERSION
+
+    export FORCE_UPDATE="false"
+    if ! check_signed; then
+        echo "You're updating to an unsigned images, you must override the verification using --force flag"
+        FORCE_UPDATE="true"
+    else
+        echo "You're updating to a signed images, so run the upgrade command without --force flag"
+    fi
+    if [[ "${FORCE_UPDATE}" == "false" ]]; then
+        admin_ack
+        cco_annotation
+    fi
+    if [[ "${UPGRADE_CCO_MANUAL_MODE}" == "oidc" ]]; then
+        update_cloud_credentials_oidc
+    fi
+    if ! check_ota_case_enabled "OCP-53907"; then
+        set_channel
```

Comment:
We already set channel in preupgrade step, we do not need this if block, right?

Reply:
not yet, only case ocp-53907 will do some checks depending on channel update. so skip set_channel for ocp-53907. for other cases in the future, they just call tolatest step for a `--to-latest` upgrade if they don't need prechecks depending on channel.

Reply:
We are setting channel in preupgrade step only when `ENABLE_OTA_TEST: OCP-53907` is setting in config level, so that means if `OCP-53907` is not enabled in other ci jobs, `set_channel` will never happen, sounds we do not need this if block, is my understanding right?

Reply:
set_channel is a required step if we want do Y-upgrade with --to-latest. if ocp-530907 is not enabled in other ci jobs, we need it to get available update due to default channel is on X.

Reply:
Ah, got it, sounds like we always need `set_channel` whatever running `OCP-53907` once tolatest step is referenced, if that, I think we remove the if condition to always run it and remove the step of setting channel in preupgrade. WDTY?

Reply:
yes, we always need set_channel. But for ocp-53907, we need set_channel in advance, which needed to be ran in preupgrade for the checkpoint on available update. It's an exception here.

Reply:
I am thinking if we can move the checkpoint in preupgrade to this step, once we always run `set_channel`, so that easily understand the case design idea.

Reply:
I also thinking about it. But finally, i prefer to keep case checkpoint in pre/post upgrade steps to get tolatest step more simple and clear. and cases specific checkpoints in pre/post upgrade steps is the initial purpose for it to toimage step.

Reply:
Okay, in this step, it is using `DUMMY_TARGET_VERSION` as target channel, while in your case, you are using source version + 1 to set target channel, though we also can  `DUMMY_TARGET_VERSION` to do that, but that would introduce more codes in preupgrade, so I am okay with the current code now.